finagle-core/src/main/scala/com/twitter/finagle/Dtab.scala

```diff
@@ -154,6 +162,106 @@ object Dentry {
       d1.dst.simplified == d2.dst.simplified
     )
   }
+
+  /**
+   * A Prefix comprises a [[Path]]-matching expression.
+   *
+   * Each element in a prefix may be either a [[Label]] or [[AnyElem]].
+   * When matching a [[Path]], Label-elements must match exactly,
+   * while Any-elements are ignored.
+   */
+  case class Prefix(elems: Prefix.Elem*) {
+
+    def matches(path: Path): Boolean = {
+      if (this.size > path.size)
+        return false
+      var i = 0
+      while (i != this.size)
+        elems(i) match {
+          case Prefix.Label(buf) if buf != path.elems(i) =>
+            return false
+          case _ => // matches
+            i += 1
+        }
+      true
+    }
+
+    // A prefix acts somewhat like a Seq[Elem]
+    def take(n: Int) = Prefix(elems.take(n):_*)
```

Comment:
I was just trying to keep this totally in line with Path api.  I can omit them if you think that would be better